buildSrc/src/main/java/OnlyAffectedConnectedTestTask.kt

```diff
@@ -0,0 +1,55 @@
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+/**
+ * This task will run only the [connectedAndroidTest] for affected modules and dependants.
+ *
+ * This task dependsOn: [https://github.com/leandroBorgesFerreira/dag-command]
+ * That will generate for us a list of affected modules
+ *
+ */
+open class OnlyAffectedConnectedTestTask : DefaultTask() {
+
+    init {
+        group = "verification"
+        description = "Installs and runs the tests for debug on connected devices (Only for affected modules)."
+
+        setDependsOn(mutableListOf("dag-command"))
+    }
+
+    @TaskAction
+    fun runOnlyAffectedConnectedTest() {
+        var affectedModules: Set<String> = mutableSetOf()
+        File("${project.buildDir}/dag-command/affected-modules.json").useLines {
+            affectedModules = it.joinToString()
+                .removeSurrounding("[", "]")
+                .replace("\"", "")
+                .split(",")
+                .toSet()
+        }
+
+        if (affectedModules.isEmpty() || affectedModules.first().isEmpty()) {
+            println("\uD83E\uDD8B It is not necessary to run any test, ending here to free up some resources.")
+            return
+        }
+
+        targetModules.forEach { module ->
+            if (affectedModules.contains(module)) {
+                val testTask = "$module:$testTaskSuffix"
+                println("\uD83D\uDD27 Running tests on '$testTask'.")
+                project.exec {
+                    args(testTask)
+                    executable("./gradlew")
+                }
+            }
+        }
+    }
+
+    private companion object {
+        // makes sense to maintain dependencies in a logical order of execution
+        private val targetModules = listOf("cryptography", "persistence", "logic")
+        private const val testTaskSuffix = "connectedAndroidTest"
```

Comment:
It shouldn't be hard to replicate this for all tests, right?

iOS, JVM and JS for example.

I guess the issue is that not all modules have iOS and JS for example. But in general I guess we can make an `abstract` task and then just change the `targetModules` and `testTaskSuffix`.

Or, maybe we can figure out if the module has iOS and JS. 🧐 
I'll leave that to future versions of us to figure out.

Reply:
Makes total sense. I was wondering the same with iOS, at least I can give it a shot to apply this also for iOS tests (our other slow brother). Since we can read configuration from proj objt, I think it's possible to get if a module has this tests or not 🤔

Reply:
Get test by name should be really simple actually.

```kotlin
tasks
    .firstOrNull { it.name.equals("iOSTest", true }
    ?.execute()
```

Reply:
Ok, made some changes, now we are reusing some code, doing some "auto" config using some Kotlin DSL goodies.

I'm not positive about applying it to jvm tests since we could mess with the coverage 🤔 . We can research later, but anyway those tests are the fastest now =) Let me know what you think :D